telemetry/prometheus_module.go

```diff
@@ -33,22 +34,31 @@ const (
 	PrometheusModuleName = "prometheus"
 )
 
-func CreatePrometheusTelemetryModule(cfg *TelemetryConfig) (*PrometheusTelemetryModule, error) {
+func CreatePrometheusTelemetryModule(runtime modules.Runtime) (modules.Module, error) {
+	var m PrometheusTelemetryModule
+	return m.Create(runtime)
+}
+
+func (m *PrometheusTelemetryModule) Create(runtime modules.Runtime) (modules.Module, error) {
+	cfg := runtime.GetConfig()
+	if err := m.ValidateConfig(cfg); err != nil {
+		log.Fatalf("config validation failed: %v", err)
```

Comment:
I don't really understand the pattern of not returning the error and logging fatal instead. (**Ditto for all module creation**)

Reply:
Addressed in https://github.com/pokt-network/pocket/pull/235/commits/2a686db94104405ac0ff6b2320c740bbf32a11ae

Reply:
Need to update this one as well